events/events.go

```diff
@@ -200,24 +200,19 @@ func (c ContainerLaunchContext) GetMicroservices() []MicroserviceSpec {
 }
 
 func (c ContainerLaunchContext) GetServicePathElement() *persistence.ServiceInstancePathElement {
-	return c.ServicePathElement
+	return &c.ServicePathElement
 }
 
 func NewContainerLaunchContext(config *ContainerConfig, envAdds *map[string]string, bc BlockchainConfig, name string, agId string, mss []MicroserviceSpec, spe *persistence.ServiceInstancePathElement) *ContainerLaunchContext {
-	c := new(ContainerLaunchContext)
-	c.Configure = *config
-	c.EnvironmentAdditions = envAdds
-	c.Blockchain = bc
-	c.Name = name
-	c.AgreementId = agId
-	c.Microservices = mss
-	if(spe != nil){
-		c.ServicePathElement = new(persistence.ServiceInstancePathElement)
-		*c.ServicePathElement = *spe
-	} else {
-		c.ServicePathElement = nil
+	return &ContainerLaunchContext{
```

Comment:
Could you add code here to guard against spe to be nil? 
When spe is nil, make `spe_temp := persistence.NewServiceInstancePathElement("", "")` and `c.ServicePathElement  = *spe_temp`.